myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+                    unit = '' if v.unit() is None else f' {v.unit()}'
+                    export_lines.append(
+                        f'{lhs} {{ {rhs} }} /* {qname}{unit} */'
+                    )
+                export_lines.append('')
+
+        # Add initial conditions `u_i`
+        export_lines.append('/* Initial conditions */')
+        export_lines.append('u_i {')
+        for v in model.states():
+            lhs = var_name(v)
+            rhs = myokit.float.str(v.initial_value(as_float=True))
+            qname = v.qname()
+            unit = '' if v.unit() is None else f' {v.unit()}'
+            export_lines.append(f'{tab}{lhs} = {rhs}, /* {qname}{unit} */')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Add initial conditions `dudt_i`
+        export_lines.append('dudt_i {')
+        for v in model.states():
+            lhs = var_name(v.rhs())  # Use intermediary dot_x instead of dot(x)
+            rhs = myokit.float.str(v.initial_value(as_float=True))
+            export_lines.append(f'{tab}{lhs} = {rhs},')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Add remaining variables
+        todo_vars = (
+            set(model.variables(const=False, deep=True, state=False))
+            - special_vars
+        )
+        for component_label, eq_list in sorted_eqs.items():
+            todo_eqs = [
+                eq for eq in eq_list.equations() if eq.lhs.var() in todo_vars
+            ]
+            if todo_eqs:
+                export_lines.append(f'/* Variables: {component_label} */')
+                for eq in todo_eqs:
+                    v = eq.lhs.var()
+                    lhs = var_name(v)
+                    rhs = e.ex(eq.rhs)
+                    qname = v.qname()
+                    unit = '' if v.unit() is None else f' {v.unit()}'
+                    export_lines.append(
+                        f'{lhs} {{ {rhs} }} /* {qname}{unit} */'
+                    )
+                export_lines.append('')
+
+        # Add `F_i`
+        export_lines.append('/* Solve */')
+        export_lines.append('F_i {')
+        for v in model.states():
+            lhs = var_name(v.rhs())  # Use intermediary dot_x instead of dot(x)
+            export_lines.append(f'{tab}{lhs},')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Add `G_i`
+        export_lines.append('G_i {')
+        for v in model.states():
+            rhs = e.ex(v.rhs().rhs())  # Use rhs of intermediary dot_x
+            export_lines.append(f'{tab}{rhs},')
+        export_lines.append('}')
+        export_lines.append('')
+
+        # Output state variables + extra output variables
+        export_lines.append('/* Output */')
+        export_lines.append('out_i {')
+        vars = set(model.states())
+        if extra_out_vars:
+            vars = vars.union(extra_out_vars)
+        for v in sorted(vars, key=lambda x: x.qname()):
```

Comment:
What's the order of the outputs here? Should probably explain that in the docstring. Can we go `states + extra-in-order-given-by-user` ?

Reply:
I'll change `extra_out_vars` to `currents` as that's what it really is. The order is alphabetical, I'll put that in the docstring.

Reply:
Resolved by 1a212493. `extra_out_vars` changed to `currents`, order made alphabetical (`key = lambda x: var_name(x).lowercase()`, and comment added in docstring.